src/boot/efi/linux.c

```diff
@@ -86,6 +86,16 @@ static EFI_STATUS load_image(EFI_HANDLE parent, const void *source, size_t len,
                         len,
                         ret_image);
 
+        /* Retry with a more hacky override. */
+        if (IN_SET(ret, EFI_ACCESS_DENIED, EFI_SECURITY_VIOLATION) && install_security_hack())
+                ret = BS->LoadImage(
+                                /*BootPolicy=*/false,
+                                parent,
+                                &payload_device_path.payload.Header,
+                                (void *) source,
+                                len,
+                                ret_image);
```

Comment:
tbh, if replacing the protocol interface doesn't work, if we should keep using it anyway. I mean, yeah, patching vtables is frickin ugly, but if the clean way doesn't work on some systems then I'd just give in. I mean, there's benefit in having a single codepath for this mess...

(But a comment explaining why we do it this way and that we explored the cleaner options would be wise)

Reply:
or to ask this question explicitly: sticking to overriding the protocol interface as first attempt, what does that actually gain us if it doesn't work on many machines?

Reply:
My main motivation is because it is ever so slightly cleaner. But that's about it. Any actual arguments in favor of reinstall protocol first are mostly theoretical at this point (other than the fuzzy feeling that we're not mucking around in stuff we don't own unless we have to).

From the reports I have a feeling it is mostly older firmware that doesn't support the reinstall method. It also wouldn't surprise me if edk2 from around the same time would also exhibit the same behavior.

Reply:
sure, it's cleaner, but if you have to keep the ugly codepath too, then what do you get? ugly code anway...